### PR TITLE
 refactor(downloader): RHICOMPL-1912 to download other files

### PR DESCRIPTION
**Pull request by vkrizan**
## Secure Coding Practices Checklist GitHub Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Checklist
- [X] Input Validation
- [X] Output Encoding
- [X] Authentication and Password Management
- [X] Session Management
- [X] Access Control
- [X] Cryptographic Practices
- [X] Error Handling and Logging
- [X] Data Protection
- [X] Communication Security
- [X] System Configuration
- [X] Database Security
- [X] File Management
- [X] Memory Management
- [X] General Coding Practices

### CODE REVIEW

**Comment by akofink** [comment]
Looking good!

File: app/services/safe_downloader.rb [hunk 1]
```diff
@@ -20,22 +20,36 @@ def initialize(message = 'report is empty')
...
+      downloaded_file
     rescue *DOWNLOAD_ERRORS => e
       raise DownloadError if e.instance_of?(RuntimeError) &&
                              e.message !~ /redirection/
 
       raise DownloadError, "download failed (#{url}): #{e.message}"
     end
 
+    def download_reports(url, opts = {})
+      opts[:ssl_only] = Rails.env.production?
```
Comment (bastilian) [inline app/services/safe_downloader.rb:41]: Should this option be exposed higher up, maybe in the general app settings as `...__DOWNLOAD_SSL`?

Comment (vkrizan) [reply app/services/safe_downloader.rb]: Refactored to be settable via `Settings.report_download_ssl_only`, that defaults to `true` for production. Thanks!

File: config/settings/production.yml [hunk 2]
```diff
@@ -2,5 +2,6 @@ host_inventory_url: http://insights-inventory.platform-prod.svc:8080
 rbac_url: http://rbac.rbac-prod.svc.cluster.local:9002
 prometheus_exporter_host: compliance-prometheus-exporter.compliance-prod.svc
 prometheus_exporter_port: 9394
+report_download_ssl_only: true
```
Comment (akofink) [inline config/settings/production.yml:5]: :+1: good call, Sebastian. Much better :)

File: app/consumers/concerns/report_parsing.rb [hunk 3]
```diff
@@ -96,7 +96,10 @@ def b64_identity
     end
 
     def report_contents
-      @report_contents ||= SafeDownloader.download(url)
+      @report_contents ||= SafeDownloader.download_reports(
+        url,
+        ssl_only: Settings.report_download_ssl_only
```
Comment (akofink) [inline app/consumers/concerns/report_parsing.rb:101]: Why pass this in when we can just check this global setting directly?

**Review by akofink — APPROVED** [review]
Nice! Thanks, @vkrizan!

File: app/consumers/concerns/report_parsing.rb (hunk 3 above)
Comment (vkrizan) [reply app/consumers/concerns/report_parsing.rb:101]: Idk, it made more sense to me to do it this way and have it as an option. :slightly_smiling_face:

**Pull request merged**